.circleci/config.yml

```
@@ -1,68 +1,132 @@
-
-              py.test
-              bash <(curl -s https://codecov.io/bash) -c
-
-            else
-              echo "Skipping"
-            fi
-
-      - add_ssh_keys
+            - venv
 
-      - deploy:
-          name: Release package
-          command: |
-
-            if [ "${CIRCLE_BRANCH}" == "master" ]; then
-
-              git config --global user.email "ci@circleci.com"
-              git config --global user.name "$CIRCLE_USERNAME"
-              bin/release
-            else
-              echo "Not releasing"
-            fi
+jobs:
+  pylama:
+    executor: python
+    steps:
+      - checkout
+      - setup_venv
+      - install_deps
+      - run: pylama
+  isort:
+    executor: python
+    steps:
+      - checkout
+      - setup_venv
+      - install_deps
+      - run: isort --check-only --recursive .
+  black:
+    executor: python
+    steps:
+      - checkout
+      - setup_venv
+      - install_deps
+      - run: black --check .
+  test:
+    executor: python
+    steps:
+      - checkout
+      - setup_venv
+      - install_deps
+      - run: pipenv run pytest --junitxml=test-reports/pytest/junit.xml
+      - run: pipenv run codecov
+  cut-release:
+    executor: python
+    parameters:
+      dryrun:
+        type: boolean
+        default: false
+    steps:
+      - checkout
+      - setup_venv
+      - install_deps
+      - run: git config user.email "admin+github@dialogue.co"
+      - run: git config user.name “CircleCI”
+      - run: semantic-release version<<# parameters.dryrun >> --noop<</ parameters.dryrun >>
+      - unless:
+          condition: << parameters.dryrun >>
+          steps:
+            - run: git push --follow-tags
+  package-and-deploy:
+    executor: python
+    steps:
+      - checkout
+      - setup_venv
+      - install_deps
+      - run: python setup.py sdist
+      - run: >
+          curl -F package=@dist/guestlist-$(python setup.py --version).tar.gz
+          https://${GEMFURY_PUSH_TOKEN}@push.fury.io/dialogue/
+      - run: semantic-release changelog --post
+  build-and-deploy-docs:
+    executor: python
+    environment:
+      GIT_DEPLOY_DIR: build/sphinx/html
+    steps:
+      - checkout
+      - setup_venv
+      - install_deps
+      - run: git config user.email "admin+github@dialogue.co"
+      - run: git config user.name “CircleCI”
+      - run: python setup.py build_sphinx
+      - run: bash bin/deploy-docs.sh
+
+workflows:
+  main:
+    jobs:
+      - pylama:
+          context: org-global
+      - isort:
+          context: org-global
+      - black:
+          context: org-global
+      - test:
+          context: org-global
+      - cut-release:
+          context: org-global
+          requires:
+            - pylama
+            - isort
+            - black
+            - test
+          filters:
+            branches:
+              only:
+                - master
+  publish:
+    jobs:
+      - package-and-deploy:
+          context: org-global
```

```suggestion
          context: org-global-v2
```
? and similarly elsewhere?

---

👍